components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/OAuthUserConsentedScopesDAO.java

```diff
@@ -0,0 +1,78 @@
+package org.wso2.carbon.identity.oauth2.dao;
+
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2ScopeConsentException;
+import org.wso2.carbon.identity.oauth2.model.UserApplicationScopeConsentDO;
+
+import java.util.List;
+
+/**
+ * This interface defines the API for user consent management for OAuth scopes.
+ */
+public interface OAuthUserConsentedScopesDAO {
+
+    /**
+     * Retrieve the user consent given for OAuth scopes for a given application.
+     *
+     * @param userId    User identifier.
+     * @param appId     Application identifier.
+     * @param tenantId  Tenant Id.
+     * @return  {@link UserApplicationScopeConsentDO}
+     * @throws IdentityOAuth2ScopeConsentException
+     */
+    UserApplicationScopeConsentDO getUserConsentForApplication(String userId, String appId, int tenantId)
```

Comment:
```suggestion
    UserApplicationScopeConsentDO getExistingUserConsentForApplication(String userId, String appId, int tenantId)
```

Reply:
Does this needed since this method is in the DAO layer?